client/web/src/global/CoolCodeIntel.tsx

```diff
@@ -369,18 +408,48 @@ const CollapsibleLocationList: React.FunctionComponent<{
                             <ChevronRightIcon className="icon-inline" aria-label="Expand" />
                         )}{' '}
                         {capitalize(props.name)}
+                        <Badge pill={true} variant="secondary" className="ml-2">
+                            {props.locations.length}
+                            {props.hasMore && '+'}
+                        </Badge>
                     </h4>
                 </Button>
             </CardHeader>
 
             <Collapse id="references" isOpen={isOpen}>
                 {props.locations.length > 0 ? (
-                    <LocationsList
-                        locations={props.locations}
-                        activeLocation={props.activeLocation}
-                        setActiveLocation={props.setActiveLocation}
-                        filter={props.filter}
-                    />
+                    <>
+                        <LocationsList
+                            locations={props.locations}
+                            activeLocation={props.activeLocation}
+                            setActiveLocation={props.setActiveLocation}
+                            filter={props.filter}
+                        />
+                        {props.hasMore &&
+                            props.fetchMore !== undefined &&
+                            (disable ? (
+                                <div className="text-center mb-1">
+                                    <em>Loading more {props.name}...</em>
+                                    <LoadingSpinner inline={true} />
+                                </div>
+                            ) : (
+                                <div className="text-center mb-1">
+                                    <Button
+                                        variant="secondary"
+                                        disabled={disable}
+                                        onClick={event => {
+                                            event.preventDefault()
+                                            setDisable(true)
+                                            if (props.fetchMore) {
```

Comment:
```suggestion
                                        onClick={()=> {
                                            setDisable(true)
                                            if (props.fetchMore) {
```

It's just type="button" by default so don't think we need to do anything here